### PR TITLE
EXOGTN-2335 Make template statistics calculation totaly async

### PR DESCRIPTION
**Pull request by boubaker**

### CODE REVIEW

File: component/scripting/src/main/java/org/exoplatform/groovyscript/text/TemplateService.java [hunk 1]
```diff
@@ -142,7 +142,9 @@ public void merge(String name, BindingContext context) throws Exception {
           final Long time = endTime - startTime;
           final TemplateStatistic templateStatistic = statisticService.getTemplateStatistic(name);
           templateStatistic.setResolver(resourceResolver);
-          statisticTimes.add(new TemplateStatisticTime(templateStatistic, time));
+          CompletableFuture.runAsync(() -> {
+            statisticTimes.add(new TemplateStatisticTime(templateStatistic, time));
+          });
```
Comment (thomasdelhomenie) [inline component/scripting/src/main/java/org/exoplatform/groovyscript/text/TemplateService.java]: Why not doing the whole block (inside if(collectTemplateStatistics_ )) asynchronous ?

Comment (boubaker) [reply component/scripting/src/main/java/org/exoplatform/groovyscript/text/TemplateService.java]: The whole bloc is not blocker for the template statistics collection. So we need to make only this operation asynchronously.

Comment (thomasdelhomenie) [reply component/scripting/src/main/java/org/exoplatform/groovyscript/text/TemplateService.java]: Of course not, but it makes more sense to me to move all the operation of collecting statistics asynchronous instead than only a subpart.

Comment (boubaker) [reply component/scripting/src/main/java/org/exoplatform/groovyscript/text/TemplateService.java]: This change is not relevant IMO, else we will need to redo the TQA tests.
The change is just making the blocking operation async, for the other statements, no need to make it async for now.

Comment (thomasdelhomenie) [reply component/scripting/src/main/java/org/exoplatform/groovyscript/text/TemplateService.java]: Not sure to get the logic. But that's not blocker, no problem.

**Pull request merged**